libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1108,12 +1108,12 @@ void CompilerUtils::convertType(
 					// Value shrank
 					for (unsigned j = targetSize; j < sourceSize; ++j)
 					{
-						moveToStackTop(depth - 1, 1);
+						moveToStackTop(depth + targetSize - sourceSize, 1);
 						m_context << Instruction::POP;
 					}
 					// Value grew
 					if (targetSize > sourceSize)
-						moveIntoStack(depth + targetSize - sourceSize - 1, targetSize - sourceSize);
+						moveIntoStack(depth - sourceSize, targetSize - sourceSize);
```

Comment:
Can anyone confirm that this makes sense? We basically want to move the ``targetSize - sourceSize`` slots past all *other* tuple components, which have ``depth - sourceSize`` slots.

I'm not aware of any code that would trigger this with ``sourceSize != 0``, though. And so far it only worked for ``targetSize == 1``.

Reply:
At the line `// Move it back into its place` we have:

` "sourceSize" original items | "depth - sourceSize" unrelated items | "targetSize" converted items`

total size: depth + targetSize

After that loop (assume targetSize > sourceSize):

` "sourceSize" modified items | "depth - sourceSize" unrelated items | "targetSize - sourceSize" converted items`

total size: depth + targetSize - sourceSize

Since targetSize > sourceSize, the situation at "value grew" is the same.

And we have to move `targetSize - sourceSize` items past a number of `depth - sourceSize` items.

So the change is correct.